### PR TITLE
Added basic angular app with Spring Boot example

### LINKED ISSUE #921: Add angular + Spring Boot example

### PR DESCRIPTION
**Pull request by dogeared**
resolves #921

### CODE REVIEW

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/filter/mvc/ControllerFilter.java [hunk 1]
```diff
@@ -117,6 +117,13 @@ protected void filter(HttpServletRequest request, HttpServletResponse response,
             return;
         }
 
+        View view = this.viewResolver.getView(vm, request);
```
Comment (dogeared) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/filter/mvc/ControllerFilter.java]: @mario @jarias - I'd like your feedback on this change. This is what makes the `stormpath.web.produces = application/json` work properly for the angular app (although, it should make it work in general, not just for angular).

However, below we originally thought that if view was null, we should be returning a 404.

NOTE: If I don't explicitly create a controller that handles `/login` (for instance), then - even with these changes - we do still get the expected 404, so I think this is ok. Just want to check with you.

Comment (jarias) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/filter/mvc/ControllerFilter.java]: Looks good

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/AbstractController.java [hunk 2]
```diff
@@ -209,7 +209,7 @@ protected boolean isJsonPreferred(HttpServletRequest request, HttpServletRespons
         try {
             return MediaType.APPLICATION_JSON.equals(contentNegotiationResolver.getContentType(request, response, produces));
         } catch (UnresolvedMediaTypeException e) {
-            log.error("Couldn't resolve content type", e);
+            log.info("isJsonPreferred: Couldn't resolve content type: {}", e.getMessage());
```
Comment (dogeared) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/mvc/AbstractController.java]: This makes sure we log a more reasonable message when content type can't be resolved.

We were treating this as an error before, but if you've explicitly configured the properties to *not* produce text/html (like the angular example), then perhaps this should only be a debug message?

What do you think @mrioan @jarias?

File: examples/spring-boot-webmvc-angular/pom.xml [hunk 3]
```diff
@@ -0,0 +1,88 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  ~ Copyright 2015 Stormpath, Inc.
```
Comment (mraible) [inline examples/spring-boot-webmvc-angular/pom.xml]: Should this be 2016?

File: examples/spring-boot-webmvc-angular/src/main/java/com/stormpath/spring/boot/examples/AngularController.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+import javax.servlet.http.HttpServletResponse;
+
+import static org.springframework.web.bind.annotation.RequestMethod.GET;
+
+@Controller
+public class AngularController {
+
```
Comment (mraible) [inline examples/spring-boot-webmvc-angular/src/main/java/com/stormpath/spring/boot/examples/AngularController.java]: Remove extra blank line.

File: examples/spring-boot-webmvc-angular/src/main/java/com/stormpath/spring/boot/examples/Application.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    @Bean
+    public WebHandler loginPostHandler() {
+        return new WebHandler() {
+            @Override
+            public boolean handle(HttpServletRequest request, HttpServletResponse response, Account account) {
+                log.debug("----> PostLoginHandler");
+                return true;
+            }
+        };
+    }
```
Comment (mraible) [inline examples/spring-boot-webmvc-angular/src/main/java/com/stormpath/spring/boot/examples/Application.java]: Are the web handlers necessary if they don't do anything?

File: examples/spring-boot-webmvc-angular/src/main/resources/static/app.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+angular.module('exampleApp', [
+  'ui.router',
+  'stormpath',
+  'stormpath.templates'
+])
+  .config(function ($stateProvider, $urlRouterProvider, $locationProvider, STORMPATH_CONFIG) {
+    $urlRouterProvider
+      .otherwise('/');
+
+    $locationProvider.html5Mode(true);
```
Comment (mraible) [inline examples/spring-boot-webmvc-angular/src/main/resources/static/app.js]: Did you get things to work with HTML5 mode? I remember this being an issue.

Comment (dogeared) [inline examples/spring-boot-webmvc-angular/src/main/resources/static/app.js]: yes! With the change to `ControllerFilter`, it properly handles HTML5 mode.

File: examples/spring-boot-webmvc-angular/src/main/java/com/stormpath/spring/boot/examples/Application.java (hunk 5 above)
Comment (dogeared) [inline examples/spring-boot-webmvc-angular/src/main/java/com/stormpath/spring/boot/examples/Application.java]: Good point. I will get rid of them - they're a distraction from the example.

File: examples/spring-boot-webmvc-angular/src/main/resources/static/index.html [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+<!--[if lt IE 7]>      <html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
+<!--[if IE 7]>         <html class="no-js lt-ie9 lt-ie8"> <![endif]-->
+<!--[if IE 8]>         <html class="no-js lt-ie9"> <![endif]-->
+<!--[if gt IE 8]><!-->
+<html class="no-js"> <!--<![endif]-->
+  <head>
+    <meta charset="utf-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <base href="/">
+    <title></title>
```
Comment (mraible) [inline examples/spring-boot-webmvc-angular/src/main/resources/static/index.html]: I'd suggest adding a title like "Stormpath Angular Example"

File: examples/spring-boot-webmvc-angular/src/main/resources/static/profile/profile.html [hunk 8]
```diff
@@ -0,0 +1,52 @@
+<div class="container">
+  <div class="row">
+    <div class="col-xs-12">
+      <h3 class="text-center">My Profile</h3>
+      <hr>
+      <div class="row">
+        <div class="col-xs-12">
+          <div class="alert alert-info">
+            This form allows you to change your first and last name, which are core properties of the Stormpth Account object. The "favorite color" field is a custom property that is saved to the Custom Data object of the Account.
```
Comment (mraible) [inline examples/spring-boot-webmvc-angular/src/main/resources/static/profile/profile.html]: Typo - should be "Stormpath", not "Stormpth".

File: examples/spring-boot-webmvc-angular/src/main/java/com/stormpath/spring/boot/examples/AngularController.java (hunk 4 above)
Comment (dogeared) [inline examples/spring-boot-webmvc-angular/src/main/java/com/stormpath/spring/boot/examples/AngularController.java]: done

**Comment by mraible** [comment]
LGTM!

**Comment by dogeared** [comment]
Passed TCK against 5 integrations. merging.

**Pull request merged**